File: dotnet/base/Mcma.Api/WorkerRequest.cs
using System.Collections.Generic;
using Mcma;

namespace Mcma.Api
{
    public class WorkerRequest
    {
        public string OperationName { get; set; }

        public IDictionary<string, string> ContextVariables { get; set; }

        public object Input { get; set; }

        public McmaTracker Tracker { get; set; }
    }
}

File: base/worker/lib/worker-request.ts
import { Logger, McmaException, McmaTrackerProperties,  } from "@mcma/core";
import { WorkerRequestProperties } from "./worker-request-properties";

export class WorkerRequest implements WorkerRequestProperties {
    operationName: string;
    input?: { [key: string]: any };
    tracker?: McmaTrackerProperties;
    logger?: Logger;

    constructor(request: WorkerRequestProperties, logger?: Logger) {
        const operationName = request?.operationName;
        if (!operationName || typeof operationName !== "string") {
            throw new McmaException("operationName must be a non-empty string.");
        }

        this.operationName = operationName;
        this.input = request.input;
        this.tracker = request.tracker;
        this.logger = logger;
    }
}
